test/explain/algorithm/test_captum_explainer_heterogeneous.py

```diff
@@ -0,0 +1,215 @@
+from typing import Optional
+
+import pytest
+import torch
+
+from torch_geometric.explain import Explainer, HeteroExplanation
+from torch_geometric.explain.algorithm import CaptumExplainer
+from torch_geometric.explain.config import (
+    MaskType,
+    ModelConfig,
+    ModelMode,
+    ModelReturnType,
+    ModelTaskLevel,
+)
+from torch_geometric.nn import HANConv, HGTConv, global_add_pool
+from torch_geometric.testing import withPackage
+
+methods = [
+    'Saliency',
+    'InputXGradient',
+    'IntegratedGradients',
+    'GuidedBackprop',
+    'ShapleyValueSampling',
+]
+
+unsupported_methods = [
+    'Deconvolution',
+    'FeatureAblation',
+    'Occlusion',
+    'DeepLift',
+    'DeepLiftShap',
+    'GradientShap',
+    'KernelShap',
+    'Lime',
+]
+
+
+class HetGraphNet(torch.nn.Module):
+    def __init__(self, model_config: ModelConfig, metadata: dict,
+                 in_channels: int = -1, base_conv: str = 'han',
+                 hid_dim: int = 8):
+        super().__init__()
+
+        self.model_config = model_config
+        self.hid_dim = hid_dim
+        if model_config.mode == ModelMode.multiclass_classification:
+            self.out_channels = 7
+        else:
+            self.out_channels = 1
+
+        if (base_conv == 'hgt'):
+            self.conv = HGTConv(in_channels, self.hid_dim, heads=2,
+                                metadata=metadata)
+            self.conv2 = HGTConv(self.hid_dim, self.out_channels, heads=1,
+                                 metadata=metadata)
+        elif (base_conv == 'han'):
+            self.conv = HANConv(in_channels, self.hid_dim, heads=2,
+                                metadata=metadata)
+            self.conv2 = HANConv(self.hid_dim, self.out_channels, heads=1,
+                                 metadata=metadata)
+
+        self.relu = torch.nn.ReLU()
+
+    def forward(self, x_dict, edge_index_dict, batch_info=None,
```

Comment:
Add typing information?

Reply:
I followed the pattern of other tests (https://github.com/pyg-team/pytorch_geometric/blob/master/test/explain/algorithm/test_captum_explainer.py) where we don't really bother to type things when they're just used for testing. Happy to add it if we want to here, though we should probably try to be consistent across the tests.